src/Backend/AcceptedJob.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-- | Type and handler for Jobs that we accept or ignored
+module Backend.AcceptedJob
+    ( AcceptedJob(..)
+    , acceptJob
+    , IgnoredJob(..)
+    )
+where
+
+import Import.NoFoundation
+
+import Backend.AcceptedWebhook
+import Backend.Marketplace
+import Control.Monad.Except
+import qualified Data.Text as T
+
+data AcceptedJob = AcceptedJob
+    { ajRepo :: Entity Repo
+    , ajJob :: Entity Job
+    }
+    deriving Show
+
+newtype IgnoredJob = IgnoredJob
+    { unIgnoredJob :: Entity Job
+    }
+    deriving Show
+
+-- | Given an Accepted webhook, accept or ignore that Job
+acceptJob :: MonadIO m => AcceptedWebhook -> ExceptT IgnoredJob m AcceptedJob
+acceptJob AcceptedWebhook {..} = do
+    when (repoSvcs repo /= GitHubSVCS) $ ignoreJob awJob $ NonGitHubRepo repo
+    whenMarketplacePlanForbids allows $ ignoreJob awJob . PlanLimitation repo
+    pure $ AcceptedJob awRepo awJob
+  where
+    repo = entityVal awRepo
+    allows = awMarketplaceAllows
+
+ignoreJob
+    :: MonadIO m => Entity Job -> IgnoredJobReason -> ExceptT IgnoredJob m ()
+ignoreJob job = throwError <=< toIgnoredJob job
```

Comment:
Pushing this into `acceptJob`'s `where` will make usage nicer and not lose anything, IMO.